src/core/shared.lua

```diff
@@ -30,5 +30,36 @@ AddEventHandler("ncs_core:trace", function(message, logType)
     _NCS:trace(message, logType)
 end)
 
+local registredEvents = {}
+local function isEventRegistred(eventName)
+    for k,v in pairs(registredEvents) do
+        if v == eventName then return true end
+    end
+    return false
+end
+
+---registerNetEvent
+---@param eventName string
+---@return void
+---@public
+function _NCS:registerNetEvent(eventName, ...)
+    if not isEventRegistred(eventName) then
+        RegisterNetEvent(self:formatEvent(eventName), ...)
+        RegisterNetEvent(eventName, function(...)
+            -- TODO: Add something to prevent wrong use of event
+            -- This can be prevent stupid cheaters try to use the event without the proper format
+        end)
+        table.insert(registredEvents, eventName)
```

Comment:
```suggestion
        registredEvents[eventName] = true
```
key-index table indexing is more efficient than adding in the table.